riscv-executor/src/lib.rs

```diff
@@ -1560,7 +1568,7 @@ impl<F: FieldElement> Executor<'_, '_, F> {
 
                 let addr = addr as u32;
                 let lid = self.instr_link_id(instr, MachineInstance::memory, 0);
-                self.proc.set_mem(addr, value.u(), self.step + 3, lid);
```

Comment:
Do we have any `mload/mstore` used directly on FE values? I would assume not, in which case it would be good to assert here that value is u32

Reply:
As it stands now, in the `riscv-runtime`, the compiler is free to move around and even `memcpy()` `OpaqueGoldilocks` values.

Reply:
right